From ccb731ddb39c58fa8c4eb09492418020b1785b05 Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Wed, 27 Mar 2013 16:34:16 -0700
Subject: [PATCH 37/96] HBASE-7596 Redundant FSTableDescriptor update

Reason: Bug
Author: Matteo Bertozzi
Ref: CDH-9889
(cherry picked from commit 734c61e812a1700ba7790d67f127f9fa051a1e54)
---
 .../master/handler/TableDeleteFamilyHandler.java   |    2 +-
 .../master/handler/TableModifyFamilyHandler.java   |    7 ++-----
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/src/main/java/org/apache/hadoop/hbase/master/handler/TableDeleteFamilyHandler.java b/src/main/java/org/apache/hadoop/hbase/master/handler/TableDeleteFamilyHandler.java
index b520762..b44f1f5 100644
--- a/src/main/java/org/apache/hadoop/hbase/master/handler/TableDeleteFamilyHandler.java
+++ b/src/main/java/org/apache/hadoop/hbase/master/handler/TableDeleteFamilyHandler.java
@@ -47,7 +47,7 @@ public class TableDeleteFamilyHandler extends TableEventHandler {
   protected void handleTableOperation(List<HRegionInfo> hris) throws IOException {
     MasterFileSystem mfs = this.masterServices.getMasterFileSystem();
     // Update table descriptor
-    mfs.deleteColumn(tableName, familyName);
+    this.masterServices.getMasterFileSystem().deleteColumn(tableName, familyName);
     // Remove the column family from the file system
     for (HRegionInfo hri : hris) {
       // Delete the family directory in FS for all the regions one by one
diff --git a/src/main/java/org/apache/hadoop/hbase/master/handler/TableModifyFamilyHandler.java b/src/main/java/org/apache/hadoop/hbase/master/handler/TableModifyFamilyHandler.java
index 34560ec..34d7cdc 100644
--- a/src/main/java/org/apache/hadoop/hbase/master/handler/TableModifyFamilyHandler.java
+++ b/src/main/java/org/apache/hadoop/hbase/master/handler/TableModifyFamilyHandler.java
@@ -47,11 +47,8 @@ public class TableModifyFamilyHandler extends TableEventHandler {
 
   @Override
   protected void handleTableOperation(List<HRegionInfo> regions) throws IOException {
-    // Update table descriptor in HDFS
-    HTableDescriptor htd =
-      this.masterServices.getMasterFileSystem().modifyColumn(tableName, familyDesc);
-    // Update in-memory descriptor cache
-    this.masterServices.getTableDescriptors().add(htd);
+    // Update table descriptor
+    this.masterServices.getMasterFileSystem().modifyColumn(tableName, familyDesc);
   }
 
   @Override
-- 
1.7.0.4

